### PR TITLE
Add auto-connected info message in dashcard mapping

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/43964

[figma](https://www.figma.com/design/mcmJekuToH3AyvL8cDhlo9/Make-Dashboard-filter-auto-wire-less-presumptuous?node-id=6085-87&t=tffBQ71RRpeqxgx9-0)

image

### Demo

https://github.com/metabase/metabase/assets/125459446/a34f4650-1246-4278-9eeb-647c31ad9829

### CODE REVIEW

File: e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-wiring.cy.spec.js [hunk 1]
```diff
@@ -84,6 +84,21 @@ describe("dashboard filters auto-wiring", () => {
         "contain",
         "The filter was auto-connected to all questions containing “User.Name”.",
       );
+
+      cy.log("verify auto-connect info is shown");
```
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-wiring.cy.spec.js:88]: Repro steps:
1. create a dashboard, add filter
2. add card to it, connect it manually to the filter
3. add the same card again
4. auto-connect this time
5. click the filter

Icon is not visible

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx [hunk 2]
```diff
@@ -137,6 +141,16 @@ export function DashCardCardParameterMapper({
   const isTemporalUnit =
     editingParameter != null && isTemporalUnitParameter(editingParameter);
 
+  const autoConnectedDashcardIds = useMemo(() => {
+    return autoConnectedUndos.flatMap(undo =>
+      undo.extraInfo ? (undo.extraInfo.dashcardIds as DashCardId[]) : [],
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx:146]: Maybe even change the typo for `Undo` to a union based on the type + some base case

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.unit.spec.jsx [hunk 3]
```diff
@@ -42,6 +42,7 @@ const setup = options => {
       dashcard={createMockDashboardCard({ card })}
       question={new Question(card, metadata)}
       editingParameter={createMockParameter()}
+      autoConnectedUndos={[]}
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.unit.spec.jsx:45]: We could add unit tests here

**Review by ranquild — APPROVED** [review]
LGTM, needs small polishing

**Review by ranquild — COMMENTED** [review]
Based on the TS error I think it currently won't work with multiple parameters connected to the same auto-connected card

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx [hunk 4]
```diff
@@ -316,48 +324,90 @@ export function DashCardCardParameterMapper({
...
+              <Tooltip label={t`Auto-connected`}>
+                <Icon name="sparkles" />
+              </Tooltip>
+            ) : null}
+          </Flex>
         </>
       )}
+      <Transition
+        mounted={
+          isRecentlyAutoConnected && layoutHeight > 3 && !!selectedMappingOption
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx]: IDK I left it but moved `isRecentlyAutoConnected && !!selectedMappingOption `

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx [hunk 5]
```diff
@@ -89,6 +90,11 @@ const mapStateToProps = (
   metadata: getMetadata(state),
   question: getQuestionByCard(state, props),
   mappingOptions: getDashcardParameterMappingOptions(state, props),
+  // @ts-expect-error rework redux/undo to ts
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx]: yup, fixed

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.unit.spec.jsx (hunk 3 above)
Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.unit.spec.jsx]: added

File: frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx [hunk 6]
```diff
@@ -89,6 +90,11 @@ const mapStateToProps = (
...
+  isRecentlyAutoConnected: getIsRecentlyAutoConnectedDashcard(
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx:93]: passed

[the remark this answers is not in the document]
Comment (uladzimirdev) [reply frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.tsx:93]: fixed here [`4ed50c4` (#43930)](https://github.com/metabase/metabase/pull/43930/commits/4ed50c40037f2053bb71fd583b2bf7e1c2318586)

**Pull request merged**